File: server/app/schemas/document.py
from datetime import datetime

from pydantic import BaseModel


class DocumentUploadResponse(BaseModel):
    filename: str
    file_size: int
    pages: int
    characters: int
    chunks: int
    message: str


class DocumentResponse(BaseModel):
    id: int
    filename: str
    file_size: int
    uploaded_at: datetime

    class Config:
        from_attributes = True

File: client/src/types/document.ts
export interface UploadedDocument {
  id: number;
  filename: string;
  file_size: number;
  uploaded_at: string;
}

export interface UploadResponse {
  chat_id: number;
  document_id: number;
  filename: string;
  file_size: number;
  pages: number;
  characters: number;
  chunks: number;
  message: string;
}